### DISCUSSION TITLE
Vercel Dev not working on mac

### DISCUSSION THREAD

**Discussion by daviddecorso** [vercel/vercel#6639] [Help] [answered]
Vercel dev command works fine on my PC, I can access the API and the frontend starts up, etc. One of our team members was having difficulties on their mac so I tried it on mine and it would start up the frontend, but never start up the API and I would eventually get the error below. When I visited the port the backend was listening on in a browser, it would load, but never do anything. The dev environments are consistent between PC and mac, so I'm not sure why this would be happening.
```
Error! Detecting port 49655 timed out after 300000ms
> [debug] [2021-08-26T16:32:56.851Z] Error: Detecting port 49655 timed out after 300000ms
    at checkForPort (/Users/daviddecorso/Document/Coding/QRDocentDashboard/node_modules/vercel/dist/index.js:234489:19)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at DevServer.runDevCommand (/Users/daviddecorso/Document/Coding/QRDocentDashboard/node_modules/vercel/dist/index.js:234296:9)
    at DevServer._start (/Users/daviddecorso/Document/Coding/QRDocentDashboard/node_modules/vercel/dist/index.js:233910:9)
    at Object.dev [as default] (/Users/daviddecorso/Document/Coding/QRDocentDashboard/node_modules/vercel/dist/index.js:225532:5)
    at main (/Users/daviddecorso/Document/Coding/QRDocentDashboard/node_modules/vercel/dist/index.js:225641:16)
    at main (/Users/daviddecorso/Document/Coding/QRDocentDashboard/node_modules/vercel/dist/index.js:210304:16)
```

**Comment by tedbyron** [comment, marked as answer]
@daviddecorso have you found a solution to this issue? I'm running the exact same configs on windows and macOS and only the windows setup works. Tried changing and checking ports on the mac, but I'm a bit stumped.

Running `vercel dev -l 8000` I can see that it's listening on the correct port but the page never loads when I open `localhost:8000`
```bash
└ $ lsof -nP -iTCP -sTCP:LISTEN | rg node
COMMAND     PID USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
node      10160  ted   26u  IPv6 0xb9dfeae5471f71bd      0t0  TCP *:8000  (LISTEN)
node      10165  ted   33u  IPv4 0xb9dfeae549d7304d      0t0  TCP *:59876 (LISTEN)
```

**Reply by daviddecorso** [reply]
Yeah, my dev command for yarn was specifying a port for the frontend and when I changed it to not specify a port it worked fine on my mac. Not sure if that'll fix your problem, but it worked for me.

**Comment by johnhwhite** [comment]
I was able to solve this issue by specifying `--host 127.0.0.1` for my dev command. It was listening on `[::1]` by default (IPv6), and the proxy expects `127.0.0.1` (IPv4).

**Comment by unitof** [comment]
@daviddecorso Just curious: is your project build on Sanity, or a different framework? I encountered this with the default Vercel `dev` command for Sanity (https://github.com/vercel/vercel/discussions/8511) and have a [fix](https://github.com/vercel/vercel/pull/8512), but am trying to figure out if this affects other frameworks as well.

**Reply by daviddecorso** [reply]
Nope, this was just plain React.

**Reply by unitof** [reply]
Gotcha! Wonder if it would be considered a `vc dev` bug then, on macOS. I'll follow up in #8511.

**Reply by unitof** [reply]
Vict'ry is nigh! https://github.com/vercel/vercel/pull/8542

**Reply by unitof** [reply]
Fix you can test out: https://github.com/vercel/vercel/issues/8121#issuecomment-1242339369
